frontend/javascripts/components/flash.js

```diff
@@ -0,0 +1,63 @@
+const Vue = require("vue");
+
+let Flash = Vue.extend({
+  template: "#js-ann-flash",
+  data: function() {
+    return {
+      type: gon.flash.type || "notice",
+      body: gon.flash.body || ""
+    };
+  },
+  computed: {
+    show: function() {
+      return !!this.body;
+    },
+    alertClass: function() {
+      switch (this.type) {
+        case "notice":
+          return "alert-success";
+          break;
+        case "info":
+          return "alert-info";
+          break;
+        case "alert":
+          return "alert-warning";
+          break;
+        case "danger":
+          return "alert-danger";
+          break;
+      }
+    },
+    alertIcon: function() {
+      switch (this.type) {
+        case "notice":
+          return "fa-check-circle";
+          break;
+        case "info":
+          return "fa-info-circle";
+          break;
+        case "alert":
+          return "fa-exclamation-circle";
+          break;
+        case "danger":
+          return "fa-exclamation-triangle";
+          break;
+      }
+    }
+  },
+  methods: {
+    close: function() {
+      this.body = "";
+    }
+  },
+  ready: function() {
+    if (this.show) {
+      let self = this;
```

Comment:
'let' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).